### PR TITLE
changes for error framework in solver

### PR DESCRIPTION
**Pull request by sachingrover211**
Complete implementation of the error framework and changes to show the error messages in demo as well.

### CODE REVIEW

File: matrix.js [hunk 1]
```diff
@@ -74,7 +74,10 @@ define([], function(){
 		var isValid = _validateMatrices(a);
 
 		if(!isValid.inv)
-			throw new Error("Matrix can not be inverted as it is not a square Matrix");
+			throw {
+				type: "decomposition",
+				message: messages.get("decomposition")
```
Comment (riteshsamala) [inline matrix.js:88]: Should't matrix.js include messages object in this file ?

Comment (sachingrover211) [reply matrix.js:88]: I didnt want it to unnecessarily hold an object, hence just exposed a function to get the things from somewhere else.

Comment (sachingrover211) [reply matrix.js:88]: This way nothing can be changed in matrix.js to show any other messages.

**Comment by sachingrover211** [comment]
@jonwetzel @riteshsamala can this pull request be merged ?

**Review by riteshsamala — APPROVED** [review]

**Comment by jonwetzel** [comment]
@dharanisakary Sachin has requested your review of this PR, so I'm making you the second reviewer.

**Review by dharanisakary — APPROVED** [review]
Tested this with solver_feedback branch of topomath

**Pull request merged**